### ISSUE TITLE
Failsafe option doesn't work for SSH Auth Failure

### ISSUE THREAD

**Issue by henokgetachew** [pstadler/flightplan#165]
I have searched for a solution to this but I haven't been able to find one.  Here's the log that flightplan fails with (I have replaced host names)
```
● Safely failed connecting to '{host1-address}': getaddrinfo ENOTFOUND {host1-address}
● Safely failed connecting to '{host2-address}': getaddrinfo ENOTFOUND {host2-address}
● Error: This Fiber is not yielding
    at Client.<anonymous> ({path-to-my-project}/node_modules/flightplan/lib/transport/ssh.js:53:18)
    at emitOne (events.js:96:13)
    at Client.emit (events.js:191:7)
    at tryNextAuth ({path-to-my-project}/node_modules/ssh2/lib/client.js:296:12)
    at SSH2Stream.onUSERAUTH_FAILURE ({path-to-my-project}/node_modules/ssh2/lib/client.js:473:5)
    at emitTwo (events.js:106:13)
    at SSH2Stream.emit (events.js:194:7)
    at parsePacket ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:3647:10)
    at SSH2Stream._transform ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:551:13)
    at SSH2Stream.Transform._read (_stream_transform.js:167:10)
    at SSH2Stream._read ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:212:15)
    at SSH2Stream.Transform._write (_stream_transform.js:155:12)
```
I have set the property `failsafe` to true for each of the hosts.  The error is coming from node-fibers ([this code](https://github.com/laverdet/node-fibers/blob/c0456ac8d1a0162abc6a10aab51d07af00f1cb76/src/fibers.cc#L549)).  I am not sure how to proceed.  Any help, including workarounds, is appreciated.  Thanks.

**Comment by henokgetachew** [comment]
Update: this only happens when the failure is an auth failure.  Here's the error that makes it crash:
```
{ Error: All configured authentication methods failed
    at tryNextAuth ({path-to-my-project}/node_modules/ssh2/lib/client.js:294:17)
    at SSH2Stream.onUSERAUTH_FAILURE ({path-to-my-project}/node_modules/ssh2/lib/client.js:473:5)
    at emitTwo (events.js:106:13)
    at SSH2Stream.emit (events.js:194:7)
    at parsePacket ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:3647:10)
    at SSH2Stream._transform ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:551:13)
    at SSH2Stream.Transform._read (_stream_transform.js:167:10)
    at SSH2Stream._read ({path-to-my-project}/node_modules/ssh2-streams/lib/ssh.js:212:15)
    at SSH2Stream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:329:12)
    at writeOrBuffer (_stream_writable.js:315:5) level: 'client-authentication' }
```

**Comment by pstadler** [comment]
Are you running the latest version? I'm unable to reproduce this.

```sh
$ node --version; fly --version
v6.9.1
0.6.15

# ---
$ cat flightplan.js
var plan = require('flightplan')

plan.target('test',
[
  {
    host: 'nohost',
    failsafe: true,
    user: 'root'
  },
  {
    host: '<host that exists>',
    port: 22,
    user: 'root',
    failsafe: true
  },
  {
    host: '<host that exists>',
    port: 22,
    user: 'root',
    failsafe: true
  },
  {
    host: 'foobarfofofo',
    failsafe: true,
    user: 'root'
  },
])

plan.remote(function (remote) {
  remote.exec('ls -al')
})

# ---
$ fly test
✈ Running default:test
✈ Connecting to 'nohost'
✈ Connecting to 'kube2.pstadler.io'
✈ Connecting to 'kube3.pstadler.io'
✈ Connecting to 'foobarfofofo'
● Safely failed connecting to 'nohost': getaddrinfo ENOTFOUND nohost nohost:22
● Safely failed connecting to 'foobarfofofo': getaddrinfo ENOTFOUND foobarfofofo foobarfofofo:22
● Safely failed connecting to '<host that exists>': All configured authentication methods failed
● Safely failed connecting to '<host that exists>': All configured authentication methods failed
✈ Flightplan finished after 463 ms
```

**Comment by henokgetachew** [comment]
You should be able to reproduce it if you add `agent: 'process.env.SSH_AUTH_SOCK'`.
```
$ node --version; fly --version
v7.8.0
0.6.15

#-----
$ cat flightplan.js
var plan = require('flightplan')

plan.target('test', [
  {
    host: 'nohost',
    failsafe: true,
    user: 'root'
  },
  {
    host: '<host that exists>',
    port: 22,
    user: 'root',
    failsafe: true
  },
  {
    host: ‘<host that exists>’,
    port: xxx,
    user: ‘my_username’,
    agent: 'process.env.SSH_AUTH_SOCK',
    password: 'xyz',
    failsafe: true
  },
  {
    host: 'foobarfofofo',
    failsafe: true,
    user: 'root'
  }
])

plan.remote(function (remote) {
  remote.exec('ls -al')
})
$ fly test
✈ Running default:test
✈ Connecting to 'nohost'
✈ Connecting to '<host that exists>'
✈ Connecting to 'acms.app.medicmobile.org'
✈ Connecting to 'foobarfofofo'
● Safely failed connecting to 'nohost': getaddrinfo ENOTFOUND nohost nohost:22
● Safely failed connecting to '<host that exists>': getaddrinfo ENOTFOUND <host that exists> <host that exists>:22
● Safely failed connecting to 'foobarfofofo': getaddrinfo ENOTFOUND foobarfofofo foobarfofofo:22
● Safely failed connecting to 'acms.app.medicmobile.org': connect ENOENT process.env.SSH_AUTH_SOCK
✈ Flightplan finished after 2.23 s
● Error: This Fiber is not yielding
    at Client.<anonymous> ({path-to-my-project}//node_modules/flightplan/lib/transport/ssh.js:53:15)
    at emitOne (events.js:96:13)
    at Client.emit (events.js:191:7)
    at tryNextAuth ({path-to-my-project}//node_modules/ssh2/lib/client.js:296:12)
    at SSH2Stream.onUSERAUTH_FAILURE ({path-to-my-project}//node_modules/ssh2/lib/client.js:473:5)
    at emitTwo (events.js:106:13)
    at SSH2Stream.emit (events.js:194:7)
    at parsePacket ({path-to-my-project}//node_modules/ssh2-streams/lib/ssh.js:3647:10)
    at SSH2Stream._transform ({path-to-my-project}//node_modules/ssh2-streams/lib/ssh.js:551:13)
    at SSH2Stream.Transform._read (_stream_transform.js:167:10)

```

**Comment by henokgetachew** [comment]
Looks as though the code's looking to get a final output while the server is giving it another chance at auth?

**Comment by pstadler** [comment]
I could reproduce this by passing `agent: 'process.env.SSH_AUTH_SOCK'`. This should be **without quotes**. Nevertheless, I released [0.6.16](https://github.com/pstadler/flightplan/releases/tag/0.6.16) to prevent such problems.

**Comment by henokgetachew** [comment]
Confirming that this works.  Thanks.
